lib/gems/pending/openstack/openstack_handle/handle.rb

```diff
@@ -197,6 +197,17 @@ def connect(options = {})
 
         raw_service = self.class.raw_connect_try_ssl(username, password, address, port, service, opts, api_version,
                                                      security_protocol)
+
+        # need to check if this is versionless keystone endpoint
+        if service == "Identity"
+          identity_prefix = (api_version == "v2" ? "v2.0" : "v3")
+          unless raw_service.credentials[:openstack_management_url].include?(identity_prefix)
+            opts[:openstack_identity_prefix] = identity_prefix
+            raw_service = self.class.raw_connect_try_ssl(username, password, address, port, service, opts, api_version,
```

Comment:
@Ladas that's the part that I couldn't figure out; the reason we run it twice is because the only I could figure out to tell if keystone is versionless, is to connect to the service and look at openstack_management_url.  Note that we only run it twice if it's versionless.